controllers/bookController.js

```diff
@@ -33,6 +70,15 @@ exports.getBook = catchAsync(async (req, res, next) => {
   });
 });
 exports.updateBook = catchAsync(async (req, res, next) => {
+  const errors = validationResult(req);
+  const id = req.params.id;
+  const findBook = await Book.findById(id);
+  if (!findBook) {
+    return next(new AppError("No book found with that ID", 404));
+  }
+  if (!errors.isEmpty()) {
```

Comment:
Error validation should come before this

  const findBook = await Book.findById(id);
  if (!findBook) {
    return next(new AppError("No book found with that ID", 404));
  }